lib/chef/knife/ec2_server_create.rb

```diff
@@ -689,6 +689,8 @@ def bootstrap_common_params(bootstrap)
         bootstrap.config[:first_boot_attributes_from_file] = locate_config_value(:first_boot_attributes_from_file)
         bootstrap.config[:encrypted_data_bag_secret] = s3_secret || locate_config_value(:secret)
         bootstrap.config[:encrypted_data_bag_secret_file] = locate_config_value(:secret_file)
+        #cl_secret needs to be set as chef checks for this key
+        Chef::Config[:knife][:cl_secret] = s3_secret if locate_config_value(:s3_secret)
```

Comment:
@btm,
yes, I will be improving the comment and code as per your suggestion and have added tests also.
When secret option is passed for Linux vm, instead of knife-windows, chef module is imported.
In chef code mentioned [here](https://github.com/chef/chef/blob/master/lib/chef/knife/data_bag_secret_options.rb#L43), when secret option is passed, its proc function sets the value using set_cl_secret method in `Chef::Config[:knife][:cl_secret]`.
And later the chef code instead of checking if value is present in `bootstrap.config` also, it checks for value in `Chef::Config[:knife][:cl_secret]` [code here](https://github.com/chef/chef/blob/master/lib/chef/knife/data_bag_secret_options.rb#L70) 
Hence only setting the s3 secret value in `bootstrap.config[:secret]` doesn't work, as only upon verification that `cl_secret` is set, the code uses `bootstrap.config[:secret]` value.